File: CreationalPatterns/Singleton/Java/ThreadSafePizzaOven.java
// Eager Initialization

public class ThreadSafePizzaOven {
    private static final ThreadSafePizzaOven oven = new ThreadSafePizzaOven();

    private ThreadSafePizzaOven(){}

    public static ThreadSafePizzaOven getOven(){
        return oven;
    }

    public void bake(){
        System.out.println("Baking in thread safe oven");
    }

    public void preHeat(){
        System.out.println("Preheating thread safe oven");
    }
}

File: CreationalPatterns/Singleton/C#/Singleton/ThreadSafePizzaOven.cs
// Eager Initialization

public class ThreadSafePizzaOven {
    private static readonly ThreadSafePizzaOven oven = new ThreadSafePizzaOven();

    private ThreadSafePizzaOven(){}

    public static ThreadSafePizzaOven GetOven(){
        return oven;
    }

    public void Bake(){
        Console.WriteLine("Baking in thread safe oven");
    }

    public void PreHeat(){
        Console.WriteLine("Preheating thread safe oven");
    }
}
